### ISSUE TITLE
ipaclient logger requires root access

### ISSUE THREAD

**Issue by briantward** [freeipa/ansible-freeipa#184]
I tried running my playbook to install ipa-client and got permissions errors when the python code on the ansible host required access to write to /var/log/ipa-client-install.log.  Ideally I would like to be able to do this with any user and not need to be root from the ansible host.  Running with root works fine.

Playbook from the example:

```
---
- name: Playbook to configure IPA clients with username/password
  hosts: ipaclients
  collections:
    - freeipa.ansible_freeipa
  roles:
  - role: ipaclient
    state: present
```

Appears to be using this library that requires root permissions to open its logger:

```
$ rpm -q --whatprovides /usr/lib/python3.7/site-packages/ipapython/ipa_log_manager.py
python3-ipalib-4.8.3-1.fc30.noarch
```

Log error:
````
$ ansible-playbook register-idm.yaml -vvv
ansible-playbook 2.9.2
  config file = /home/bward/projects/personal/example.com/ansible.cfg
  configured module search path = ['/home/bward/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.7.5 (default, Oct 17 2019, 12:09:47) [GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]
Using /home/bward/projects/personal/example.com/ansible.cfg as config file
SSH password: 
Parsed /home/bward/projects/personal/example.com/inventory inventory source with ini plugin

PLAYBOOK: register-idm.yaml **********************************************************************************************************************
1 plays in register-idm.yaml

PLAY [Playbook to configure IPA clients with username/password] **********************************************************************************

..[omissions]...

TASK [ipaclient : Install - IPA client test] *****************************************************************************************************
task path: /home/bward/.ansible/collections/ansible_collections/freeipa/ansible_freeipa/roles/ipaclient/tasks/install.yml:33
<XXX.XXX.XXX.XXX> ESTABLISH SSH CONNECTION FOR USER: root
<XXX.XXX.XXX.XXX> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o PubKeyAuthentication=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/home/bward/.ansible/cp/9e40a9fb55 XXX.XXX.XXX.XXX '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<XXX.XXX.XXX.XXX> (0, b'/root\n', b'X11 forwarding request failed\r\n')
<XXX.XXX.XXX.XXX> ESTABLISH SSH CONNECTION FOR USER: root
<XXX.XXX.XXX.XXX> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o PubKeyAuthentication=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/home/bward/.ansible/cp/9e40a9fb55 XXX.XXX.XXX.XXX '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1578797464.1890244-20217732296697 `" && echo ansible-tmp-1578797464.1890244-20217732296697="` echo /root/.ansible/tmp/ansible-tmp-1578797464.1890244-20217732296697 `" ) && sleep 0'"'"''
<XXX.XXX.XXX.XXX> (0, b'ansible-tmp-1578797464.1890244-20217732296697=/root/.ansible/tmp/ansible-tmp-1578797464.1890244-20217732296697\n', b'X11 forwarding request failed\r\n')
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/ansible/executor/task_executor.py", line 146, in run
    res = self._execute()
  File "/usr/lib/python3.7/site-packages/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python3.7/site-packages/ansible/plugins/action/normal.py", line 46, in run
    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))
  File "/usr/lib/python3.7/site-packages/ansible/plugins/action/__init__.py", line 803, in _execute_module
    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)
  File "/usr/lib/python3.7/site-packages/ansible/plugins/action/__init__.py", line 197, in _configure_module
    **become_kwargs)
  File "/usr/lib/python3.7/site-packages/ansible/executor/module_common.py", line 1258, in modify_module
    environment=environment)
  File "/usr/lib/python3.7/site-packages/ansible/executor/module_common.py", line 1095, in _find_module_utils
    py_module_cache, zf)
  File "/usr/lib/python3.7/site-packages/ansible/executor/module_common.py", line 731, in recursive_finder
    [os.path.join(*py_module_name[:-idx])])
  File "/usr/lib/python3.7/site-packages/ansible/executor/module_common.py", line 650, in __init__
    self.get_source()
  File "/usr/lib/python3.7/site-packages/ansible/executor/module_common.py", line 665, in get_source
    pkg = import_module(self._package_name)
  File "/usr/lib64/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "/usr/lib/python3.7/site-packages/ansible/utils/collection_loader.py", line 112, in load_module
    mod = self._find_module(fullname, None, load=True)[1]
  File "/usr/lib/python3.7/site-packages/ansible/utils/collection_loader.py", line 252, in _find_module
    exec(code_object, newmod.__dict__)
  File "/home/bward/.ansible/collections/ansible_collections/freeipa/ansible_freeipa/plugins/module_utils/ansible_ipa_client.py", line 264, in <module>
    filemode='a', console_format='%(message)s')
  File "/usr/lib/python3.7/site-packages/ipapython/ipa_log_manager.py", line 85, in standard_logging_setup
    file_handler = logging.FileHandler(filename, mode=filemode)
  File "/usr/lib64/python3.7/logging/__init__.py", line 1087, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/lib64/python3.7/logging/__init__.py", line 1116, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
PermissionError: [Errno 13] Permission denied: '/var/log/ipaclient-install.log'

fatal: [mynewhost.example.com]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}

PLAY RECAP ***************************************************************************************************************************************
mynewhost.example.com    : ok=4    changed=0    unreachable=0    failed=1    skipped=4    rescued=0    ignored=0   

```

**Comment by briantward** [comment]
I thought this originally a problem with missing `become: true` in the playbook, but I've added that, and it still doesn't work.  It is definitely trying to "test load" the python script on the ansible-host prior to sending it over to the remote host.  It doesn't appear to do anything with the file it is trying to open, however, as the timestamp on it is quite old (it's been a year or more since registering the ansible host to my IPA server).  

logs for successful run as root from ansible host:

```
<XXX.XXX.XXX.XXX> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/id_rsa_btw"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="bward"' -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/932a404f38 XXX.XXX.XXX.XXX '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/bward/.ansible/tmp/ansible-tmp-1578951145.718273-38144896305711 `" && echo ansible-tmp-1578951145.718273-38144896305711="` echo /home/bward/.ansible/tmp/ansible-tmp-1578951145.718273-38144896305711 `" ) && sleep 0'"'"''
<XXX.XXX.XXX.XXX> (0, b'ansible-tmp-1578951145.718273-38144896305711=/home/bward/.ansible/tmp/ansible-tmp-1578951145.718273-38144896305711\n', b'X11 forwarding request failed\r\n')
Using module file /root/.ansible/collections/ansible_collections/freeipa/ansible_freeipa/plugins/modules/ipaclient_test.py
<XXX.XXX.XXX.XXX> PUT /root/.ansible/tmp/ansible-local-138150xt88s50/tmp265mz6pt TO /home/bward/.ansible/tmp/ansible-tmp-1578951145.718273-38144896305711/AnsiballZ_ipaclient_test.py
<XXX.XXX.XXX.XXX> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/id_rsa_btw"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="bward"' -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/932a404f38 '[XXX.XXX.XXX.XXX]'
<XXX.XXX.XXX.XXX> (0, b'sftp> put /root/.ansible/tmp/ansible-local-138150xt88s50/tmp265mz6pt /home/bward/.ansible/tmp/ansible-tmp-1578951145.718273-38144896305711/AnsiballZ_ipaclient_test.py\n', b'')
```

**Comment by jesmg** [comment]
I had the same error and did some debugging. 
The problem seems to be that the log file does not exists, and the role is trying to open it without creating before.

It also happens with ipaclient.

The solution is to ensure the file are in "present" state as an installation requirement. 
I will do a PR as soon as I have time.

**Comment by t-woerner** [comment]
This should be fixed with d31a132a595b298092242c09394c1a55a46ef0e2 already. There was an issue with the handling of collections in Ansible, that will be fixed also on the Ansible side.
